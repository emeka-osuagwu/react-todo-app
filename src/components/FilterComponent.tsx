import React from "react";
import * as RB from "react-bootstrap";

function Component({ searchFilter, statusFilter }: any) {
    return (
        <RB.Form>
            <RB.Row>
                <RB.Col sm={6}>
                    <RB.Form.Group controlId="exampleRB.Form.ControlInput1">
                        <RB.Form.Label>keyword...</RB.Form.Label>
                        <RB.Form.Control
                            onChange={(event: any) =>
                                searchFilter(event.target.value)
                            }
                            type="text"
                            placeholder="keyword..."
                        />
                    </RB.Form.Group>
                </RB.Col>
                <RB.Col sm={6}>
                    <RB.Form.Group controlId="exampleRB.Form.ControlSelect1">
                        <RB.Form.Label>Example select</RB.Form.Label>
                        <RB.Form.Control
                            onChange={(event: any) =>
                                statusFilter(event.target.value)
                            }
                            as="select"
                        >
                            <option>Select</option>
                            <option value={1}>Yes</option>
                            <option value={0}>No</option>
                        </RB.Form.Control>
                    </RB.Form.Group>
                </RB.Col>
            </RB.Row>
        </RB.Form>
    );
}

export default Component;
