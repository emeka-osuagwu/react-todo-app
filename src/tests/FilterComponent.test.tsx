import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import "@testing-library/jest-dom/extend-expect";

import FilterComponent from "../components/FilterComponent";

test("renders without crashing", () => {
    render(<FilterComponent />);
    const title = screen.getByText("Search:");
    expect(title).toBeInTheDocument();
});

test("renders same mock up", () => {
    const tree = renderer.create(<FilterComponent />).toJSON();
    expect(tree).toMatchSnapshot();
});
