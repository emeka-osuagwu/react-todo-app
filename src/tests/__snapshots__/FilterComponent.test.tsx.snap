// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders same mock up 1`] = `
<form
  className=""
  style={
    Object {
      "marginBottom": 30,
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-2"
        >
          <div
            className="form-group"
          >
            <h5
              style={
                Object {
                  "marginTop": 6,
                }
              }
            >
              Search:
            </h5>
          </div>
        </div>
        <div
          className="col-sm-10"
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="exampleRB.Form.ControlInput1"
              onChange={[Function]}
              placeholder="keyword..."
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-6"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-3"
        >
          <div
            className="form-group"
          >
            <h5
              style={
                Object {
                  "marginTop": 6,
                }
              }
            >
              Completed:
            </h5>
          </div>
        </div>
        <div
          className="col-sm-9"
        >
          <div
            className="form-group"
          >
            <select
              className="form-control"
              id="exampleRB.Form.ControlSelect1"
              onChange={[Function]}
            >
              <option>
                Select
              </option>
              <option
                value={1}
              >
                Yes
              </option>
              <option
                value={0}
              >
                No
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
  </div>
</form>
`;
